cobbler/actions/mkloaders.py

```diff
@@ -0,0 +1,212 @@
+                self.logger.info('syslinux version 4 detected! Skip making symlink of "ldlinux.c32" file!')
+                continue
+            symlink(target, link, skip_existing=True)
+
+    def make_grub(self):
+        """
+        Create symlink of the GRUB 2 bootloader in case it is available on the system. Additionally build the loaders
+        for other architectures if the modules to do so are available.
+        """
+        symlink(
+            pathlib.Path("/usr/share/efi/x86_64/grub.efi"),
+            self.bootloaders_dir.joinpath(pathlib.Path("grub/grub.efi")),
+            skip_existing=True
+        )
+
+        if not utils.command_existing("grub2-mkimage"):
+            self.logger.info("grub2-mkimage command not available. Bailing out of GRUB2 generation!")
+            return
+
+        for image_format, options in self.boot_loaders_formats.items():
+            bl_mod_dir = options.get("mod_dir", image_format)
+            mod_dir = self.grub2_mod_dir.joinpath(bl_mod_dir)
+            if not mod_dir.exists():
+                self.logger.info(
+                    'GRUB2 modules directory for arch "%s" did no exist. Skipping GRUB2 creation',
+                    image_format
+                )
+                continue
+            try:
+                mkimage(
+                    image_format,
+                    self.bootloaders_dir.joinpath("grub", options["binary_name"]),
+                    self.modules + options.get("extra_modules", []),
+                )
+            except subprocess.CalledProcessError:
+                self.logger.info('grub2-mkimage failed for arch "%s"! Maybe you did forget to install the grub modules '
+                                 'for the architecture?', image_format)
+                utils.log_exc()
+                # don't create module symlinks if grub2-mkimage is unsuccessful
+                continue
+            self.logger.info('Successfully built bootloader for arch "%s"!', image_format)
+
+            # Create a symlink for GRUB 2 modules
+            # assumes a single GRUB can be used to boot all kinds of distros
+            # if this assumption turns out incorrect, individual "grub" subdirectories are needed
+            symlink(
+                mod_dir,
+                self.bootloaders_dir.joinpath("grub", bl_mod_dir),
+            )
+
+    def create_directories(self):
+        """
+        Create the required directories so that this succeeds. If existing, do nothing. This should create the tree for
+        all supported bootloaders, regardless of the capabilities to symlink/install/build them.
+        """
+        if not self.bootloaders_dir.exists():
+            raise FileNotFoundError("Main bootloader directory not found! Please create it yourself!")
+
+        grub_dir = self.bootloaders_dir.joinpath("grub")
+        if not grub_dir.exists():
+            grub_dir.mkdir(mode=0o644)
```

Comment:
[nitpick] When I read the function name and docstring I expected more than just a creation of the "grub" directory. What about renaming the function to reflect that it only creates this one directory? E.g. `create_grub_subdirectory`

Reply:
I did try to write it in the docstring. I expect that this function is used in the future for more directories and thus I like to keep it generic. If that's fine for you please resolve the discussion. If you think otherwise I'll rename and then resolve the discussion.

Reply:
Just keep it, it's a nitpick for a reason :)